config/config.go

```diff
@@ -195,6 +204,74 @@ type Adapter struct {
 	Disabled bool `mapstructure:"disabled"`
 }
 
+// validateAdapterEndpoint validates to make sure that an adapter has a valid endpoint
+// associated with it
+func validateAdapterEndpoint(endpoint string, adapterName string, errs configErrors) configErrors {
+	if endpoint == "" {
+		errs = append(errs, fmt.Errorf("There's no default endpoint available for %s. Calls to this bidder/exchange will fail. "+
+			"Please set adapters.%s.endpoint in your app config", adapterName, adapterName))
+	} else {
+		// Check if endpoint has macros. If it does then resolve with dummy values
+		// for validation
+		if macros.ContainsMacro(endpoint) {
+			endpointTemplate, err := macros.BuildTemplate(endpoint)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("Invalid endpoint template: %s. %s", endpoint, err.Error()))
```

Comment:
I have changed that to `fmt.Errorf("Error parsing template: %s", err.Error())` but not totally sure if that's any better. Open to suggestions on this one

Reply:
oh... no, I was referring to the two lines together.

These two statements are equivalent:

```
errs = append(errs, err)
return errs
```

and

```
return append(errs, err)
```

Your `fmt.Errorf` was fine either way. Only suggestion I'd have there is to use `%v`  and `err` . as  the arg, rather than `%s` and `err.Error()`

Reply:
Ah ok! Looks like I misread your previous comment. Sorry, my bad!